# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1291333687 18000
#      Thu Dec 02 18:48:07 2010 -0500
# Node ID 262c365eb71c607cba202be82954344ce95c0e36
# Parent  c9df571efe958c1e32bf2d06705164f29cee5ed6
fill.m: Allow multiple filled polygons.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,11 @@
 2010-12-02  Ben Abbott <bpabbott@mac.com>
 
+	* plot/fill.m: Allow multiple filled polygons.
 	* plot/subplot.m: Add suppport for "align" and "replace" options.
 
 2010-11-26  John W. Eaton  <jwe@octave.org>
 
 	* pkg/pkg.m: Append directories to EXEC_PATH instead of
 	prepending them.  Use pathsep instead of ":".
 
 	* miscellaneous/copyfile.m, miscellaneous/movefile.m,
diff --git a/scripts/plot/fill.m b/scripts/plot/fill.m
--- a/scripts/plot/fill.m
+++ b/scripts/plot/fill.m
@@ -31,29 +31,36 @@ function retval = fill (varargin)
 
   htmp = [];
   iargs = __find_patches__ (varargin{:});
 
   oldh = gca ();
   unwind_protect
     axes (h);
 
+    nextplot = get (h, "nextplot");
+    if (strncmp (nextplot, "replace", 7))
+      set (h, "nextplot", "add");
+    endif
     for i = 1 : length (iargs)
       if (i == length (iargs))
         args = varargin (iargs(i):end);
       else
         args = varargin (iargs(i):iargs(i+1)-1);
       endif
       newplot ();
       [tmp, fail] = __patch__ (h, args{:});
       if (fail)
         print_usage();
       endif
       htmp (end + 1) = tmp;
     endfor
+    if (strncmp (nextplot, "replace", 7))
+      set (h, "nextplot", nextplot);
+    endif
   unwind_protect_cleanup
     axes (oldh);
   end_unwind_protect
 
   if (nargout > 0)
     retval = htmp;
   endif
 
